unit/deserialize.go

```diff
@@ -43,36 +43,83 @@ var (
 	ErrLineTooLong = fmt.Errorf("line too long (max %d bytes)", SYSTEMD_LINE_MAX)
 )
 
-// Deserialize parses a systemd unit file into a list of UnitOption objects.
+// Deserialize parses a systemd unit file into a list of UnitOptions.
 func Deserialize(f io.Reader) (opts []*UnitOption, err error) {
-	lexer, optchan, errchan := newLexer(f)
+	_, options, err := DeserializeAll(f)
+	return options, err
+}
+
+// DeserializeSections deserializes into a list of UnitSections.
+func DeserializeSections(f io.Reader) ([]*UnitSection, error) {
+	sections, _, err := DeserializeAll(f)
+	return sections, err
+}
+
+// DeserializeAll deserializes into UnitSections and UnitOptions.
+func DeserializeAll(f io.Reader) ([]*UnitSection, []*UnitOption, error) {
+
+	lexer, secchan, optchan, errchan := newLexer(f)
```

Comment:
(Here and everywhere else) Nit: I'd prefer naming those `sectCh` and `optCh`.

Reply:
I was following the established code formats. I can change everywhere though, if you still want this.